zookeeper/source/zk_dn_client.cc

```diff
@@ -84,64 +84,81 @@ namespace zkclient{
 
 		LOG(INFO) << "About to create the znode for the data ip";
 		if (zk->exists(BLOCK_LOCATIONS + std::to_string(uuid), exists, error_code)) {
-        	if (exists) {
-        		// TODO: Add to spec the location to which we're writing the size
-        		// zk->set(BLOCK_LOCATIONS + std::to_string(uuid), ZKWrapper::EMPTY_VECTOR, error_code); 
+			if (exists) {
+				// TODO: Add to spec the location to which we're writing the size
+				// zk->set(BLOCK_LOCATIONS + std::to_string(uuid), ZKWrapper::EMPTY_VECTOR, error_code);
 				// Add this datanode as the block's location in block_locations
 				DataNodeZNode znode_data;
 				strcpy(znode_data.ipPort, id.c_str());
 				std::vector<std::uint8_t> data(sizeof(znode_data));
```

Comment:
Sorry I confused myself, I see how the id is composed of ip:port, anyways, we still need to write that data to the payload like I had it. in the zk_nn_client they have "        id->set_xferport(payload->xferPort);
        id->set_infoport(50020);
        id->set_ipcport(payload->ipcPort);"